ximage/tests/patch/test_labels_patch.py

```diff
@@ -0,0 +1,631 @@
+    assert np.array_equal(returned_variable_array, array)
+
+    # Check with Dataset
+    variable_name = "variable"
+    dataset = xr.Dataset({variable_name: data_array})
+    returned_variable_array = labels_patch._get_variable_arr(dataset, None, centered_on)
+    assert returned_variable_array is None
+
+    returned_variable_array = labels_patch._get_variable_arr(dataset, variable_name, centered_on)
+    assert np.array_equal(returned_variable_array, array)
+
+    # Check unspecified variable with centered_on
+    centered_on = "max"
+    with pytest.raises(ValueError):
+        labels_patch._get_variable_arr(dataset, None, centered_on)
+
+
+def test_check_variable_arr():
+    """Test _check_variable_arr"""
+
+    shape = (2, 3)
+
+    # Check valid variable array (same shape as label arary)
+    variable_array = np.random.rand(*shape)
+    label_array = np.random.randint(1, 5, shape)
+    assert labels_patch._check_variable_arr(variable_array, label_array) is variable_array
+
+    # Check invalid variable array (different shape as label arary)
+    variable_array = np.random.rand(3, 3)
+    with pytest.raises(ValueError):
+        labels_patch._check_variable_arr(variable_array, label_array)
+
+
+def test_get_point_centroid():
+    """Test _get_point_centroid"""
+
+    shape = (4, 5)
+    array = np.random.rand(*shape)
+    expected_centroid = tuple(np.array(shape) / 2)
+    assert labels_patch._get_point_centroid(array) == expected_centroid
+
+    # Check None return if only nan in array
+    array = np.full(shape, np.nan)
+    assert labels_patch._get_point_centroid(array) is None
+
+
+def test_get_point_random():
+    """Test _get_point_random"""
+
+    shape = (4, 5)
+    array = np.random.rand(*shape)
+    returned_point = labels_patch._get_point_random(array)
+    assert len(returned_point) == len(shape)
+    for max_value, coordinate in zip(shape, returned_point):
+        assert 0 <= coordinate
+        assert coordinate < max_value
+
+    # Check None return if only nan in array
+    array = np.full(shape, np.nan)
+    assert labels_patch._get_point_random(array) is None
+
+
+def test_get_point_with_max_value():
+    """Test _get_point_with_max_value"""
+
+    shape = (4, 5)
+    array = np.random.rand(*shape)
+    max_value = np.max(array)
+
+    returned_point = labels_patch._get_point_with_max_value(array)
+    assert array[returned_point] == max_value
+
+    # Check None return if only nan in array
+    array = np.full(shape, np.nan)
+    assert labels_patch._get_point_with_max_value(array) is None
+
+
+def test_get_point_with_min_value():
+    """Test _get_point_with_min_value"""
+
+    shape = (4, 5)
+    array = np.random.rand(*shape)
+    min_value = np.min(array)
+
+    returned_point = labels_patch._get_point_with_min_value(array)
+    assert array[returned_point] == min_value
+
+    # Check None return if only nan in array
+    array = np.full(shape, np.nan)
```

Comment:
Here capture the RuntimeWarning: All-NaN slice encountered point = np.argwhere(arr == np.nanmax(arr)) so to have a clean the warning summary of the CI.
Or we can capture the warning already in the code, since we allow this behaviour ;)